### PR TITLE
[CP-1231] Refactor handlers in service desktop

### PR DESCRIPTION
**Pull request by pawel-mudita**
**Description**

Refactored handlers of sys::messages and
some cleanups in the service desktop.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

File: module-services/service-desktop/ServiceDesktop.cpp [hunk 1]
```diff
@@ -111,268 +136,244 @@ auto ServiceDesktop::requestLogsFlush() -> void
     }
 }
 
-sys::ReturnCodes ServiceDesktop::InitHandler()
+auto ServiceDesktop::getSerialNumber() const -> std::string
 {
+    return settings->getValue(std::string("factory_data/serial"), settings::SettingsScope::Global);
```
Comment (SP2FET) [inline module-services/service-desktop/ServiceDesktop.cpp:141]: please use constexpr for those string data, apply for next entries too

**Review by mjan-dev — APPROVED** [review]
As there is a lot of moved logic, please test those changes thoroughly, especially for update.

**Review by kkleczkowski — APPROVED** [review]

File: module-services/service-desktop/ServiceDesktop.cpp [hunk 2]
```diff
@@ -52,44 +31,90 @@ ServiceDesktop::~ServiceDesktop()
...
 }
 
-auto ServiceDesktop::getDeviceUniqueId() const -> std::string
+std::string ServiceDesktop::prepareBackupFilename()
 {
-    return settings->getValue(sdesktop::DeviceUniqueIdName);
+    std::array<char, 64> backupFileName{};
+    std::time_t now;
+    std::time(&now);
+    std::strftime(backupFileName.data(), backupFileName.size(), "%FT%OH%OM%OSZ", std::localtime(&now));
```
Comment (SP2FET) [inline module-services/service-desktop/ServiceDesktop.cpp:99]: maybe it's worth moving the format text to constexpr and write unit test for it?

**Review by SP2FET — APPROVED** [review]

**Comment by pawel-mudita** [comment]
I've tested these changes using tests located in `test/pytest/service-desktop`.
Communication with MC works correctly.
Update tested with `test_updater.py`

File: module-services/service-desktop/ServiceDesktop.cpp (hunk 2 above)
Comment (pawel-mudita) [reply module-services/service-desktop/ServiceDesktop.cpp:99]: For now, I'm living this as is, but I'll consider changing this in the next iteration of service-desktop refactor.

**Pull request merged**